qiskit_machine_learning/circuit/library/raw_feature_vector.py

```diff
@@ -65,19 +65,24 @@ class RawFeatureVector(BlueprintCircuit):
     def __init__(self, feature_dimension: Optional[int]) -> None:
         """
         Args:
-            feature_dimension: The feature dimension and number of qubits.
+            feature_dimension: The feature dimension from which the number of
+                                qubits is inferred as ``n_qubits = log2(feature_dim)``
 
         """
         super().__init__()
 
-        self._num_qubits = None
         self._ordered_parameters = ParameterVector("x")
-
+        self.num_qubits = None
         if feature_dimension:
             self.feature_dimension = feature_dimension
 
     def _build(self):
         super()._build()
+        num_qubits = np.log2(len(self._ordered_parameters[:]))
```

Comment:
The reason for using order parameters here rather than the instance feature dimension is because this is always set and no if None is needed? Since its set off feature dimension to start with then if the above is correct maybe add a comment for future readers of the code.

Reply:
There's a comment added in the sources.

Reply:
I looked at the comment. I am still missing something I guess. If feature dimension is None what is len(ordered_parameters[:]). If that is 0, which one might think would be a valid value if num_qubits was not known, then that will be just as much as problem giving that to np.log2.

Reply:
I spent sometime looking into `BlueprintCircuit` which I, honestly, hate, and realized that this additional check in `RawFeatureVector._build()` is not required anymore, hence `log2` is not required. The reason is that we call `super()._build()` first, which, in turns, calls `_check_configuration()` where parameters are tuned to the correct size and `RawFeatureVector._build()` works correctly afterwards.